compiler/lib/code.ml

```diff
@@ -783,18 +789,13 @@ let fold_closures_outermost_first { start; blocks; _ } f accu =
 let eq p1 p2 =
   p1.start = p2.start
   && Addr.Map.cardinal p1.blocks = Addr.Map.cardinal p2.blocks
```

Comment:
Is it useful to check  Addr.Map.cardinal first ? (now that the code uses `Map.equal`)

Reply:
Probably not

Reply:
Can you do the change and merge the PR ?